its/ruling/src/test/resources/expected/python-S5720.json

```diff
@@ -0,0 +1,67 @@
+{
+'project:django-2.2.3/django/db/utils.py':[
+247,
```

Comment:
this one looks like FP. It's a "private" method. I think we could exclude all methods starting with `_`. @nicolas-harraudeau-sonarsource what do you think?

```
def _router_func(action): ...

...

db_for_read = _router_func('db_for_read')
db_for_write = _router_func('db_for_write')
```

Reply:
Given how often there are "private" methods in python this would reduce the value of this rule quite a lot. Functions such as this one cannot be called outside of the class. I suggest to avoid raising on any function which is called in the body of the class (i.e. not in another method).
example:
``` python
In [2]: class A:
   ...:     def instance(self):
   ...:         print(self)
   ...:     def strange():   # Ok
   ...:         print("strange")
   ...:         return 42
   ...:     foo = strange()  # called here
   ...:
   ...:
strange

In [6]: a = A()

In [8]: a.strange()  # "strange" cannot be called on an instance of A
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-8-506a4d801479> in <module>
----> 1 a.strange()

TypeError: strange() takes 0 positional arguments but 1 was given
```

Reply:
ok, fine for me